sensor/common/networkflow/manager/purger.go

```diff
@@ -0,0 +1,167 @@
+			cutOff := timestamp.Now().Add(-maxAge)
+			if cutOff.After(status.tsAdded) {
+				flowMetrics.PurgerEvents.WithLabelValues("hostEndpoint", "max-age-reached").Inc()
+				delete(conns.endpoints, endpoint)
+				numPurgedEps++
+			}
+		}
+	}
+	for conn, status := range conns.connections {
+		// Remove if the related container is not found (but keep historical)
+		_, found, _ := store.LookupByContainerID(conn.containerID)
+		if !found {
+			delete(conns.connections, conn)
+			flowMetrics.PurgerEvents.WithLabelValues("hostConnection", "containerID-gone").Inc()
+			numPurgedConns++
+			continue
+		}
+		if maxAge > 0 {
+			// finally, remove all that didn't get any update from collector for a given time
+			cutOff := timestamp.Now().Add(-maxAge)
+			if cutOff.After(status.tsAdded) {
+				flowMetrics.PurgerEvents.WithLabelValues("hostConnection", "max-age-reached").Inc()
+				delete(conns.connections, conn)
+				numPurgedConns++
+			}
+		}
+	}
+	return numPurgedEps, numPurgedConns
+}
+
+func purgeActiveEndpoints(mutex *sync.Mutex, maxAge time.Duration, activeEndpoints map[containerEndpoint]*containerEndpointIndicatorWithAge, store EntityStore) int {
+	timer := prometheus.NewTimer(flowMetrics.ActiveEndpointsPurgerDuration.WithLabelValues("activeEndpoints"))
+	defer timer.ObserveDuration()
+	return concurrency.WithLock1(mutex, func() int {
+		log.Debug("Purging active endpoints")
+		return purgeActiveEndpointsNoLock(maxAge, activeEndpoints, store)
+	})
+}
+
+func purgeActiveEndpointsNoLock(maxAge time.Duration,
+	endpoints map[containerEndpoint]*containerEndpointIndicatorWithAge,
+	store EntityStore) int {
+	numPurged := 0
+	for endpoint, age := range endpoints {
+		// Remove if the endpoint is not in the store (also not in history)
+		if len(store.LookupByEndpoint(endpoint.endpoint)) == 0 {
+			delete(endpoints, endpoint)
+			numPurged++
+			flowMetrics.PurgerEvents.WithLabelValues("activeEndpoint", "endpoint-gone").Inc()
+			continue
+		}
+		// Remove if the related container is not found (but keep historical)
+		_, found, _ := store.LookupByContainerID(endpoint.containerID)
+		if !found {
+			delete(endpoints, endpoint)
+			numPurged++
+			flowMetrics.PurgerEvents.WithLabelValues("activeEndpoint", "containerID-gone").Inc()
+			continue
+		}
+		if maxAge > 0 {
+			// finally, remove all that didn't get any update from collector for a given time
+			cutOff := timestamp.Now().Add(-maxAge)
+			if cutOff.After(age.lastUpdate) {
+				flowMetrics.PurgerEvents.WithLabelValues("activeEndpoint", "max-age-reached").Inc()
+				delete(endpoints, endpoint)
+				numPurged++
+			}
+		}
+	}
+	return numPurged
+}
+
+func purgeActiveConnections(mutex *sync.Mutex, maxAge time.Duration, activeConnections map[connection]*networkConnIndicatorWithAge, store EntityStore) int {
+	timer := prometheus.NewTimer(flowMetrics.ActiveEndpointsPurgerDuration.WithLabelValues("activeConnections"))
+	defer timer.ObserveDuration()
```

Comment:
I think this is duplicated in `purgeActiveConnectionsNoLock`

Reply:
Correct! Good catch!